gitlab-pipeline/stage/test.yml

```diff
@@ -623,20 +614,41 @@ test_full_integration_enterprise:
 
     # Output storage io stats
     - df -h . | tail -1 | awk '{system("hdparm -tT "$1);}'
+
     # Get and install the integration test requirements
     - apk add $(cat ${WORKSPACE}/integration/tests/requirements/apk-requirements.txt)
-    - pip install  -r ${WORKSPACE}/integration/tests/requirements/python-requirements.txt
-    - pip3 install -r ${WORKSPACE}/integration/tests/requirements/python-requirements.txt
+    - pip install -r ${WORKSPACE}/integration/tests/requirements/python-requirements.txt
 
-    # Load all docker images
-    - for repo in `integration/extra/release_tool.py -l docker`; do
+        # Load all docker images, and the client images depending on $BUILD_CLIENT
+    - |-
+        for repo in `integration/extra/release_tool.py -l docker`; do
+        # Only load the client in the case it is built for this pipe
+            if echo $repo | grep -q mender-client; then
+                if [ "${BUILD_CLIENT}" = "true" ]; then
+                    docker load -i stage-artifacts/${repo}.tar;
+                else
+                    continue
+                fi
+            fi
         docker load -i stage-artifacts/${repo}.tar;
-      done
+        done
+
     # Login for private repos
     - docker login -u menderbuildsystem -p ${DOCKER_HUB_PASSWORD}
     - docker login -u ntadm_menderci -p ${REGISTRY_MENDER_IO_PASSWORD} registry.mender.io
     # Set testing versions to PR
-    - for repo in `integration/extra/release_tool.py -l docker`; do
+    - |-
+      for repo in `integration/extra/release_tool.py -l docker`; do
+        if echo $repo | grep -q mender-client; then
+            if [ "${BUILD_CLIENT}" = "true" ]; then
+              integration/extra/release_tool.py --set-version-of $repo --version pr;
+            else
+              # Don't set the version for the client, or set the required version?
+              # Let's try n set the required version
+              integration/extra/release_tool.py --set-version-of $repo --version "$MENDER_REV"
+              continue
```

Comment:
This should not be necessary, docker-compose files in integration repo will already have the correct version (master, 2.5.x, etc)

Reply:
Great!